libexec/copier/copier.c

```diff
@@ -698,7 +719,14 @@ main(int argc, char *argv[])
 	if (getuid() == 0 || geteuid() == 0)
 		WARNING("running as root");
 
+	signal(SIGINT, signal_handler);
+	signal(SIGTERM, signal_handler);
+
 	if (tsdfx_copier(argv[0], argv[1], maxsize) != 0)
-		exit(1);
-	exit(0);
+		retval = 1;
+
+	signal(SIGTERM, SIG_DFL);
```

Comment:
This part is pointless unless you are also going to resend the signal (which is sometimes done so the parent is informed of the fact that the child was terminated by a signal, even though it was caught and handled gracefully)